designs/2025-base-path-in-config-objects/README.md

```diff
@@ -0,0 +1,89 @@
+
+## Detailed Design
+
+Proof of concept:
+
+- `@eslint/config-array` changes: https://github.com/mdjermanovic/rewrite/pull/1
+- `eslint` changes: https://github.com/mdjermanovic/eslint/pull/6
+
+Most of the changes will be in the `@eslint/config-array` package.
+
+### Changes in `@eslint/config-array`
+
+- `baseSchema` will be updated to allow the new `basePath` property in config objects.
+- `filesAndIgnoresSchema` will be updated to validate that the new `basePath` property, if present, has a string value.
+- `normalizeConfigPatterns()` will be updated to normalize the `basePath` property value to an equivalent namespace-prefixed path, for consistency with the config array's `basePath` property value.
+- `META_FIELDS` will be updated to include `"basePath"`, in order to treat config objects with `basePath` + `ignores` as global ignores.
+- `get ignores()` will be updated to return an array of config objects instead of an array of ignore patterns.
+- `getConfigWithStatus()` will be updated to use config objects' `basePath` when present.
+- `shouldIgnorePath()` will be updated to take an array of config objects instead of ignore patterns, and use config objects' `basePath` when present.
+
+### Changes in `eslint`
+
+- In `lib/config/flat-config-array.js`, `META_FIELDS` will be updated to include `"basePath"`, in order to treat config objects with `basePath` + `ignores` as global ignores while preprocessing config array to remove global ignores when the `--no-ignore` option is used.
+- In `lib/config/config-loader.js`, the code that handles `ignorePatterns` (`--ignore-pattern`) will be updated to add them to the config array with `basePath` set to `cwd`, without any transformations.
+
+## Documentation
+
+The documentation for this feature will be added to the [Configuration Files](https://eslint.org/docs/latest/use/configure/configuration-files) page.
+
+## Drawbacks
+
+This add more complexity to the `@eslint/config-array` package.
+
+## Backwards Compatibility Analysis
+
+### `eslint`
+
+There will be no breaking changes in the `eslint` package.
+
+### `@eslint/config-array`
+
+- Introducing a new predefined property of config objects is a breaking change by itself for this package, as it can't be used as a custom property anymore.
+- `get ignores()` will have a new signature - a different return value.
+
+Neither of these changes are breaking for the `eslint` package as it doesn't use these properties.
+
+## Alternatives
+
+The `--ignore-pattern` / `ignorePatterns` problem could _maybe_ be solved by transforming patterns. However, even if it is theoretically possible, it would be immensely complex to implement correctly.
+
+## Open Questions
+
+Should this RFC include the `--basePath` feature (https://github.com/eslint/eslint/issues/19118)? I didn't see a relation between config-level base paths and ability to override config array's base path, so I didn't include it in this RFC.
```

Comment:
The reason I thought it might be good to include this feature in this RFC is because of how `--base-path` might interact with a config `basePath` property. Would it just replace the config file base path completely such that any relative `basePath` property would now be relative to the value passed as `--base-path`?

Reply:
Clarified in https://github.com/eslint/rfcs/pull/131/commits/080145fcf365874ca3402049dd7f33c6d9dd9ab6